lib/graphql/stitching/executor.rb

```diff
@@ -212,15 +212,16 @@ def initialize(supergraph:, plan:, variables: {}, nonblocking: false)
         @dataloader = GraphQL::Dataloader.new(nonblocking: nonblocking)
       end
 
-      def perform(document=nil)
+      def perform(document=nil, raw: false)
         exec!
 
-        # run the shaper...
-
         result = {}
         result["data"] = @data if @data && @data.length > 0
         result["errors"] = @errors if @errors.length > 0
-        result
+
+        result if raw
```

Comment:
Raw is a bit redundant with `document` argument. I'd say just run the shaper if a document was provided, otherwise return the raw result.

Reply:
i did not notice that document was unused there. good point. switching